client/app/widgets/Comments/__tests__/Comments.spec.jsx

```diff
@@ -83,41 +82,34 @@ const component = <Comments formProps={formProps} />;
 
 describe('Comments', () => {
   beforeEach(() => {
-    axiosPostSpy = jest.spyOn(axios, 'post').mockImplementation(() => Promise.resolve({
+    jest.spyOn(axios, 'post').mockImplementation(() => Promise.resolve({
       data: { comment },
     }));
-    axiosDeleteSpy = jest.spyOn(axios, 'delete').mockImplementation(() => Promise.resolve({
+    jest.spyOn(axios, 'delete').mockImplementation(() => Promise.resolve({
       data: { id },
     }));
   });
 
   it('renders correctly', () => {
-    let wrapper = null;
-    expect(() => {
-      wrapper = mount(component);
-    }).not.toThrow();
-    expect(wrapper).not.toBeNull();
+    render(component);
+    expect(() => component).not.toThrow();
+    expect(component).not.toBeNull();
   });
 
   it('add and delete a comment', async () => {
-    const wrapper = mount(component);
-    expect(wrapper.find('.comment').exists()).toEqual(false);
-    act(() => {
-      wrapper
-        .find('input[name="comment[comment]"]')
-        .simulate('change', { currentTarget: { value } });
+    render(component);
+    expect(screen.queryByRole('article')).toBeNull();
 
-      wrapper.find('select#comment_visibility').prop('onChange')({
-        currentTarget: { value: 'private' },
-      });
+    act(() => {
```

Comment:
Is this needed? I think `act` can be removed here 👍

Reply:
Seems like we don't need it. I still don't understand when to use `act`. I thought we change state multiple times in three events so that was why I used act. I think the purpose is to make sure assertion after all the state update is done right? Or is it because we've already used `await` and `waitFor` to wait until comment is present so there is no need of `act`?

Reply:
>I still don't understand when to use `act`

Yeah for sure it's confusing.

>the purpose is to make sure assertion after all the state update is done right?

Exactly, it flushes batched changes that may occur during async code outside of React's awareness.

>is it because we've already used `await` and `waitFor`

Right, I think it's because RTL's `waitFor` comes wrapped in `act` already, so it takes care of that for us. From what I know `act` is covered when working within RTL's async methods.

I'm going to be checking out [this article](https://kentcdodds.com/blog/fix-the-not-wrapped-in-act-warning) as I haven't read it yet. It seems to go in-depth on the use of `act` in specific scenarios. If you get a chance to go through it too (or if you already have), lmk if it makes sense 👍